docs/releases.md

```diff
@@ -1,39 +1,157 @@
-# go-ipfs releases
+# `go-ipfs` Release Flow
 
-## Release Schedule
-go-ipfs is on a six week release schedule. Following a release, there will be
-five weeks for code of any type (features, bugfixes, etc) to be added. After
-the five weeks is up, a release candidate is tagged and only important bugfixes
-will be allowed up to release day.
+## Table of Contents
 
-## Release Candidate Checklist
-- [ ] CHANGELOG.md has been updated
-  - use `./bin/mkreleaselog` to generate a nice starter list
+- [Release Philosophy](#release-philosophy)
+- [Release Flow](#release-flow)
+- [Performing a Release](#performing-a-release)
+- [Release Version Numbers (aka semver)](#release-version-numbers-aka-semver)
+
+## Release Philosophy
+
+`go-ipfs` aims to have release every six weeks, two releases per quarter. During these 6 week releases, we go through 4 different stages of Release Candidates (RC) that gives us the opportunity to test the new version against our test envinronments, production envinronments, IPFS apps (e.g. Desktop and WebUI) and with our close partners that have IPFS running in Production, by leveraging their own test infrastructure.
```

Comment:
> close partners

This terminology gives a sense of gatekeeping or a private test programme, e.g. what are the criteria to be considered a _close partner_ in the eyes of the open community? Who decides who receives that qualification, and how? I'd advocate to shift the terminology here and establish a transparent process:

1. Maybe call this "early testers programme"?
2. Maintain an `EARLY_TESTERS.md` file which is a table of Github handles to be notified when a release is available to be tested.
3. Addition and removal is by self-selection via PR on that file. We add those people to a Github team, so we can notify them at once.
4. When we produce a new release and want to subject it to early testing, we open an issue with the release notes, and mention the group.
5. Folks report their findings back on that issue, or by opening new issues referencing the _early testing_ issue.
6. We can pin those _early testing_ issues for visibility.

---

nit: typo in 'envinronments' x2

Reply:
Big thumbs up for the rename and making it clear of what are the steps to join that program

Reply:
We also could have a mailing list for pre-releases and final releases. This would be useful for package maintainers that would like to be notified about the release so they can update package in a distribution.